pkg/apis/clusterlink.net/v1alpha1/instance_types.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StatusConditionType represents the status conditions type for ClusterLink components.
```

Comment:
nit: in the future consider grouping related (e.g., Status, Spec) structures together. Also, consider if defining top-down or bottom up (i.e., is the contained struct defined before or after its containing struct).

Reply:
You are absolutely right, I usually do top-down, the Kubebuilder convention is button-up, and that's what case the mix in the definition.